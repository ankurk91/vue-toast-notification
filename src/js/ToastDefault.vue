<template>
  <div
    role="alert"
    :class="['v-toast__item', `v-toast__item--${type}`]"
  >
    <div class="v-toast__icon"></div>
    <p class="v-toast__text" v-html="message"></p>
  </div>
</template>

<script>
import {defineComponent} from 'vue';

export default defineComponent({
  name: 'Toast',
  props: {
    message: {
      type: String,
      required: true
    },
    type: {
      type: String,
      default: 'success'
    },
  },
})
</script>
