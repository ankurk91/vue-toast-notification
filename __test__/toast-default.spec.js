import {mount} from '@vue/test-utils';
import ToastDefault from '../src/js/ToastDefault.vue';

describe('ToastDefault component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(ToastDefault, {
      propsData: {
        message: 'Test message'
      },
      global: {
        stubs: {
          transition: false
        },
      }
    });
  });

  test('renders', async () => {
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.$el).toMatchSnapshot()
  });
});
