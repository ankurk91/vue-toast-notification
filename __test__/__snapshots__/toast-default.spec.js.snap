// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastDefault component renders 1`] = `
<div
  class="v-toast__item v-toast__item--success"
  role="alert"
>
  <div
    class="v-toast__icon"
  />
  <p
    class="v-toast__text"
  >
    Test message
  </p>
</div>
`;
